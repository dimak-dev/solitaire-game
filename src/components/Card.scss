.card {
    width: 100%;
    height: 100%;
    display: inline-block;
    border-radius: 1rem;
    box-shadow: 1px 1px 1px 1px black;
    //border-style: solid;
    border-width: 1px;
    background: white;
    border-color: black;
    color: black;
    font-size: 3.5vw;

    &.red {
        color: red;
    }

    .flex-wrapper {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }

    .top-line {
        flex: 1;
    }

    .bottom-line {
        flex: 1;
        rotate: 180deg;
    }
}
