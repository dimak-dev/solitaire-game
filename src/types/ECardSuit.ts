/**
 * Enumeration of playing card suites.
 *
 * Values were previously obfuscated for decreasing cheating with devtools.
 */
export enum ECardSuit {
    DIAMOND = 'gUnw0pH2E7',
    CLUB = 'mtSk8H9gkh',
    HEART = 'sGxm9srzb2',
    SPADE = 'g6N2RYyLdf',
}