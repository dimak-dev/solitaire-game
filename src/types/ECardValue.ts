/**
 * Enumeration of playing card values.
 *
 * Values were previously obfuscated for decreasing cheating with devtools.
 */
export enum ECardValue {
    ACE = 'ry4RBhALfp',
    ONE = 'ft2meneEvA',
    TWO = 'rqqHztNSTK',
    THREE = '3Np8ndDRvY',
    FOUR = 'pbpuMgeW2K',
    FIVE = 'DEUDTax2ph',
    SIX = 'ad9CpwgFL2',
    SEVEN = 'JF42qgpdC3',
    EIGHT = 'N5XLKDTQrH',
    NINE = '7ZK2SeverJ',
    JACK = 'EEYmTfXMEx',
    QUEEN = 'bBEZ2TnDrV',
    KING = 'xmXPdGVDTn',
}