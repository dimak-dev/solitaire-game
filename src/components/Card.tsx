import React, {memo} from 'react';
import './Card.scss';
import {ECardSuit} from "Types/ECardSuit";
import {ECardValue} from "Types/ECardValue";
import cardValueToChar from "Utils/cardValueToChar";
import cardSuitToChar from "Utils/cardSuitToChar";
import {cardSuiteColor} from "Utils/cardSuiteColor";


interface ICardProps {
    suit: ECardSuit,
    value: ECardValue,
}

/**
 * Card component.
 *
 * Memoized for decreasing count of renders.
 */
// const Card = memo();

export default function Card({suit, value}: ICardProps) {
    const valueChar = cardValueToChar(value);
    const suitChar = cardSuitToChar(suit);
    const color = cardSuiteColor(suit);

    return (
        <div className={`card ${color}`}>
            <div className="flex-wrapper">
                <div className="top-line">
                    {valueChar}{suitChar}
                </div>
                <div className="bottom-line">
                    {valueChar}{suitChar}
                </div>
            </div>
        </div>
    )
};