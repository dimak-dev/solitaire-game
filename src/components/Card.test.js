import {render, screen} from '@testing-library/react';
import renderer from 'react-test-renderer';
import Card from './Card';
import {ECardSuit} from "Types/ECardSuit";
import {ECardValue} from "Types/ECardValue";

test('Snapshot of Card Component', () => {
    const suits = [
        ECardSuit.CLUB,
        ECardSuit.SPADE,
        ECardSuit.HEART,
        ECardSuit.DIAMOND,
    ];

    const values = [
        ECardValue.ACE,
        ECardValue.ONE,
        ECardValue.TWO,
        ECardValue.THREE,
        ECardValue.FOUR,
        ECardValue.FIVE,
        ECardValue.SIX,
        ECardValue.SEVEN,
        ECardValue.EIGHT,
        ECardValue.NINE,
        ECardValue.JACK,
        ECardValue.QUEEN,
        ECardValue.KING,
    ];

    suits.forEach(suit => {
       values.forEach(value => {
           const tree = renderer
               .create(<Card suit={suit} value={value}/>)
               .toJSON();
           expect(tree).toMatchSnapshot();
       });
    });
});