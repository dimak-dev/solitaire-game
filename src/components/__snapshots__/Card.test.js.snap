// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Card Component 1`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      A
      ♣
    </div>
    <div
      className="bottom-line"
    >
      A
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 2`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      1
      ♣
    </div>
    <div
      className="bottom-line"
    >
      1
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 3`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      2
      ♣
    </div>
    <div
      className="bottom-line"
    >
      2
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 4`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      3
      ♣
    </div>
    <div
      className="bottom-line"
    >
      3
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 5`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      4
      ♣
    </div>
    <div
      className="bottom-line"
    >
      4
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 6`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      5
      ♣
    </div>
    <div
      className="bottom-line"
    >
      5
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 7`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      6
      ♣
    </div>
    <div
      className="bottom-line"
    >
      6
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 8`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      7
      ♣
    </div>
    <div
      className="bottom-line"
    >
      7
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 9`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      8
      ♣
    </div>
    <div
      className="bottom-line"
    >
      8
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 10`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      9
      ♣
    </div>
    <div
      className="bottom-line"
    >
      9
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 11`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      J
      ♣
    </div>
    <div
      className="bottom-line"
    >
      J
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 12`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      Q
      ♣
    </div>
    <div
      className="bottom-line"
    >
      Q
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 13`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      K
      ♣
    </div>
    <div
      className="bottom-line"
    >
      K
      ♣
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 14`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      A
      ♠
    </div>
    <div
      className="bottom-line"
    >
      A
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 15`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      1
      ♠
    </div>
    <div
      className="bottom-line"
    >
      1
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 16`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      2
      ♠
    </div>
    <div
      className="bottom-line"
    >
      2
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 17`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      3
      ♠
    </div>
    <div
      className="bottom-line"
    >
      3
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 18`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      4
      ♠
    </div>
    <div
      className="bottom-line"
    >
      4
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 19`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      5
      ♠
    </div>
    <div
      className="bottom-line"
    >
      5
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 20`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      6
      ♠
    </div>
    <div
      className="bottom-line"
    >
      6
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 21`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      7
      ♠
    </div>
    <div
      className="bottom-line"
    >
      7
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 22`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      8
      ♠
    </div>
    <div
      className="bottom-line"
    >
      8
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 23`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      9
      ♠
    </div>
    <div
      className="bottom-line"
    >
      9
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 24`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      J
      ♠
    </div>
    <div
      className="bottom-line"
    >
      J
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 25`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      Q
      ♠
    </div>
    <div
      className="bottom-line"
    >
      Q
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 26`] = `
<div
  className="card black"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      K
      ♠
    </div>
    <div
      className="bottom-line"
    >
      K
      ♠
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 27`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      A
      ♥
    </div>
    <div
      className="bottom-line"
    >
      A
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 28`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      1
      ♥
    </div>
    <div
      className="bottom-line"
    >
      1
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 29`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      2
      ♥
    </div>
    <div
      className="bottom-line"
    >
      2
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 30`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      3
      ♥
    </div>
    <div
      className="bottom-line"
    >
      3
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 31`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      4
      ♥
    </div>
    <div
      className="bottom-line"
    >
      4
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 32`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      5
      ♥
    </div>
    <div
      className="bottom-line"
    >
      5
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 33`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      6
      ♥
    </div>
    <div
      className="bottom-line"
    >
      6
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 34`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      7
      ♥
    </div>
    <div
      className="bottom-line"
    >
      7
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 35`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      8
      ♥
    </div>
    <div
      className="bottom-line"
    >
      8
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 36`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      9
      ♥
    </div>
    <div
      className="bottom-line"
    >
      9
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 37`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      J
      ♥
    </div>
    <div
      className="bottom-line"
    >
      J
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 38`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      Q
      ♥
    </div>
    <div
      className="bottom-line"
    >
      Q
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 39`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      K
      ♥
    </div>
    <div
      className="bottom-line"
    >
      K
      ♥
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 40`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      A
      ♦
    </div>
    <div
      className="bottom-line"
    >
      A
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 41`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      1
      ♦
    </div>
    <div
      className="bottom-line"
    >
      1
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 42`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      2
      ♦
    </div>
    <div
      className="bottom-line"
    >
      2
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 43`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      3
      ♦
    </div>
    <div
      className="bottom-line"
    >
      3
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 44`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      4
      ♦
    </div>
    <div
      className="bottom-line"
    >
      4
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 45`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      5
      ♦
    </div>
    <div
      className="bottom-line"
    >
      5
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 46`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      6
      ♦
    </div>
    <div
      className="bottom-line"
    >
      6
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 47`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      7
      ♦
    </div>
    <div
      className="bottom-line"
    >
      7
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 48`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      8
      ♦
    </div>
    <div
      className="bottom-line"
    >
      8
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 49`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      9
      ♦
    </div>
    <div
      className="bottom-line"
    >
      9
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 50`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      J
      ♦
    </div>
    <div
      className="bottom-line"
    >
      J
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 51`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      Q
      ♦
    </div>
    <div
      className="bottom-line"
    >
      Q
      ♦
    </div>
  </div>
</div>
`;

exports[`Snapshot of Card Component 52`] = `
<div
  className="card red"
>
  <div
    className="flex-wrapper"
  >
    <div
      className="top-line"
    >
      K
      ♦
    </div>
    <div
      className="bottom-line"
    >
      K
      ♦
    </div>
  </div>
</div>
`;
